service/src/main/java/org/databiosphere/workspacedataservice/service/RecordOrchestratorService.java

```diff
@@ -284,6 +289,49 @@ public void deleteRecordType(UUID collectionId, String version, RecordType recor
         user -> user.deleted().table().ofQuantity(1).withRecordType(recordType));
   }
 
+  public ResponseEntity<DeleteRecordsResponseServerModel> deleteRecords(
+      UUID collectionId,
+      RecordType recordType,
+      DeleteRecordsRequestServerModel deleteRecordsRequestServerModel) {
+
+    DeleteRecordsResponseServerModel response = new DeleteRecordsResponseServerModel();
+
+    Boolean hasRecordIds = !deleteRecordsRequestServerModel.getRecordIds().isEmpty();
+    Boolean hasExcludedRecordIds =
+        !deleteRecordsRequestServerModel.getExcludedRecordIds().isEmpty();
+
+    if (hasRecordIds && hasExcludedRecordIds) {
+      throw new ResponseStatusException(
+          HttpStatus.BAD_REQUEST, "record_ids and excluded_record_ids are mutually exclusive");
+    }
+
+    Boolean deleteAll = deleteRecordsRequestServerModel.getDeleteAll();
+    if (Boolean.TRUE.equals(deleteAll) && (hasRecordIds || hasExcludedRecordIds)) {
```

Comment:
throughout this method, consider using `boolean` instead of `Boolean`. That should allow you to simplify here:
```suggestion
    boolean deleteAll = deleteRecordsRequestServerModel.getDeleteAll();
    if (deleteAll && (hasRecordIds || hasExcludedRecordIds)) {
```